elastic/kibana
# fix(slo): Optimistic updates

## 🌮  Summary

This PR removes the broken optimistic updates we had in our various react query mutations for SLOs, namely useCreateSlo, useUpdateSlo, useResetSlo and useDeleteSlo. We are removing them for a few reasons:
1. useCreateSlo optimistic update was not used since we wait for the response before creating the rule and then redirecting to the list page.
2. useDeleteSlo optimistic update was broken, and deleted SLO was not removed until refresh of the page.
3. useUpdateSlo was not updating the page neither.
4. With the groupBy feature, the optimistic updates require more complexity (more requests to optimistically update/manage)

I've also notice usage of a useMutation for a non mutation API (_inspect), therefore I've changed the hook to use a useQuery instead. I had to change how this useMutation was then use in the InspectSlo flyout component.

I've also updated the EquivalentAPIRequest to handle the edit form flow, and show the correct API verb and URI.

## 🧪  Testing

- Create some SLOs
- Update some SLOs
- Delete some SLOs
- Inspect an SLO while creating
- Inspect an SLO while updating
- Check API Equivalent while creating
- Check API Equivalent while updating
